From 4fba8928c367ca823b22bccaa2cd6732893a5e41 Mon Sep 17 00:00:00 2001
From: Demon000 <demonsingur@gmail.com>
Date: Thu, 11 Jan 2018 10:56:34 +0200
Subject: [PATCH 12/19] legacy: add back imscm support into ims

Removed in 1b6c07a6c089abc3a038f308c93fd9c6934f9828,
but all pre-O blobs still need it.
Merge it into ims.

Change-Id: I3842c606af60361626d6cb78ce2690040314f625
---
 legacy-common/file_contexts   | 1 +
 legacy-common/ims.te          | 8 ++++++++
 legacy-common/platform_app.te | 3 +++
 legacy-common/service.te      | 1 +
 4 files changed, 13 insertions(+)
 create mode 100644 legacy-common/platform_app.te
 create mode 100644 legacy-common/service.te

diff --git a/legacy-common/file_contexts b/legacy-common/file_contexts
index 8ac20c6..18a04ab 100644
--- a/legacy-common/file_contexts
+++ b/legacy-common/file_contexts
@@ -9,6 +9,7 @@
 /data/misc/location/mq/alarm_svc                                    u:object_r:location_socket:s0
 
 # IMS
+/(vendor|system/vendor)/bin/imscmservice                            u:object_r:ims_exec:s0
 /dev/socket/ims_rtpd                            u:object_r:ims_socket:s0
 
 # Port bridge
diff --git a/legacy-common/ims.te b/legacy-common/ims.te
index facd4b7..d174759 100644
--- a/legacy-common/ims.te
+++ b/legacy-common/ims.te
@@ -1,5 +1,13 @@
 # Allow binder communication
 binder_use(ims)
+allow ims ims_service:service_manager add;
+allow ims imsrcs_service:service_manager add;
 
 # Allow NDC command run
 allow ims shell_exec:file rx_file_perms;
+
+# Allow communication with apps
+binder_call(ims, system_app)
+binder_call(ims, platform_app)
+
+allow ims devpts:chr_file rw_file_perms;
diff --git a/legacy-common/platform_app.te b/legacy-common/platform_app.te
new file mode 100644
index 0000000..1f11d6a
--- /dev/null
+++ b/legacy-common/platform_app.te
@@ -0,0 +1,3 @@
+# Allow binder communication with ims
+binder_call(platform_app, ims)
+allow platform_app ims_service:service_manager find;
diff --git a/legacy-common/service.te b/legacy-common/service.te
new file mode 100644
index 0000000..ee2e994
--- /dev/null
+++ b/legacy-common/service.te
@@ -0,0 +1 @@
+type ims_service, system_api_service, service_manager_type;
-- 
2.7.4

